woocommerce/payment-gateway/payment-tokens/class-sv-wc-payment-gateway-payment-token.php

```diff
@@ -538,6 +538,81 @@ private function read( \WC_Payment_Token $core_token ) {
 	}
 
 
+	/**
+	 * Stores the token data in the database.
+	 *
+	 * Stores the token data as a Woocommerce payment token, in the `wp_woocommerce_payment_tokens` table,
+	 * with meta data in the `wp_woocommerce_payment_tokenmeta` table.
+	 *
+	 * @since 5.6.0-dev.1
+	 */
+	public function save() {
+
+		if ( $this->token instanceof \WC_Payment_Token ) {
+			$token = $this->token;
+		} else {
+
+			// see if there is already a token with this ID saved for this customer and gateway
+			// TODO: use `get_user_id` and `get_gateway_id`, once merged {DM 2019-12-12}
+			$saved_tokens = \WC_Payment_Tokens::get_customer_tokens( $this->data['user_id'], $this->data['gateway_id'] );
+
+			foreach ( $saved_tokens as $saved_token ) {
+
+				if ( $saved_token->get_token() === $this->get_id() ) {
+					$token = $saved_token;
+					break;
+				}
+			}
+
+			if ( empty( $token ) ) {
+
+				// instantiate a new token
+				if ( $this->is_credit_card() ) {
+					$token = new \WC_Payment_Token_CC();
+				} elseif ( $this->is_echeck() ) {
+					$token = new \WC_Payment_Token_ECheck();
+				}
+
+				// mark the token as migrated
+				$this->data['migrated'] = true;
+
+				// update legacy token in user meta data
+				$gateways   = WC()->payment_gateways->payment_gateways();
```

Comment:
```suggestion
				$gateways   = WC()->payment_gateways()->payment_gateways();
```

Reply:
Can you please clarify why?

Reply:
WC does not define a `payment_gateways` property; instead it uses a magic method `__get()` to assign it to `payment_gateways()` method. It is harder for editors and static analyzers understand this pattern and may mark the property as non-existing; also the return type would not be clear. Instead, calling the method directly is more transparent. It's easier to point where  a problem might be when properties or methods are declared explicitly than dynamically via `__call` and `__get` in my opinion. Since you are on PTO, if you don't mind I would take the initiative to change this before merging. If you want discuss this later, please do so.